pip/src/interpreter.rs

```diff
@@ -333,15 +333,25 @@ impl Interpreter {
         Circuit(self.interpreter.get_circuit()).into_py(py)
     }
 
-    #[pyo3(signature=(entry_expr=None, callback=None))]
+    #[pyo3(signature=(entry_expr=None, callback=None, pauli_noise=None))]
     fn run(
         &mut self,
         py: Python,
         entry_expr: Option<&str>,
         callback: Option<PyObject>,
+        pauli_noise: Option<(f64, f64, f64)>,
     ) -> PyResult<PyObject> {
         let mut receiver = OptionalCallbackReceiver { callback, py };
-        match self.interpreter.run(&mut receiver, entry_expr) {
+
+        let noise = match pauli_noise {
+            None => None,
+            Some((px, py, pz)) => match PauliNoise::from_probabilities(px, py, pz) {
+                Ok(noise_struct) => Some(noise_struct),
+                Err(error_message) => return Err(PyException::new_err(error_message)),
+            },
+        };
```

Comment:
This looks ideal to me, but since you asked (in the chat), I think this is how you'd write it with `.map` etc:

```rust
let noise = pauli_noise
    .map(|(px, py, pz)| {
        PauliNoise::from_probabilities(px, py, pz).map_err(PyException::new_err)
    })
    .transpose()?;
```

Yours looks a lot cleaner.

Reply:
Thank you! I'll keep my code.